### PR TITLE
Create OpsWorks section.

### LINKED ISSUE #123: Start section: OpsWorks
Basics around when/how it's useful and a few tips, to get this section started.

### PR DESCRIPTION
**Pull request by ClydeMachine**
Started the OpsWorks section with some high-level basics, a couple well-known alternatives, and some limitations. Fixes #123.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1666,6 +1667,31 @@ Certificate Manager
 -       🔸In order to use **Certificate Manager** for CloudFront distributions certificate must be issued or imported from us-east-1 (N. Virginia) region. Certificates from other regions can [only be used with Elastic Load Balancers](https://docs.aws.amazon.com/acm/latest/userguide/acm-services.html).
 -       🔸**IoT** has its [own way](http://docs.aws.amazon.com/iot/latest/developerguide/create-device-certificate.html) of setting up certificates.
 
+OpsWorks
+-------------------
+
+### OpsWorks Basics
+
+-	📒 [Homepage](https://aws.amazon.com/opsworks/) ∙ [Documentation](https://aws.amazon.com/documentation/opsworks/) ∙ [FAQ](https://aws.amazon.com/opsworks/faqs/) ∙ [Pricing - Stacks](https://aws.amazon.com/opsworks/stacks/pricing/) ∙ [Pricing - ChefAutomate](https://aws.amazon.com/opsworks/chefautomate/pricing/)
+- OpsWorks is a configuration management service relying heavily on Chef for configuration-as-code deployment automation. The service lets you configure and launch stacks specific to your application's needs.
```
Comment (ThanosBaskous) [inline README.md]: non-blocking nit (phrasing): "service relying heavily on" -> "service that relies heavily on"

File: README.md [hunk 2]
```diff
@@ -1666,6 +1667,31 @@ Certificate Manager
...
+- There are numerous options, in and out of AWS, that let you automate application deployments. The separating factor between OpsWorks and other configuration management services (Elastic Beanstalk for example) is that OpsWorks specializes in letting you control the details of the systems your application runs on, where a service like Elastic Beanstalk simplifies this to focus on application configuration.
+- Simply put, developers who are familiar with system administration and server configuration will find great freedom with customizing their deployments by using OpsWorks.
```
Comment (ThanosBaskous) [inline README.md]: I don't feel like this bullet adds a lot - additionally, "Simply put" and "great freedom" feel out of place to me. I'd suggest losing the whole bullet.

File: README.md [hunk 3]
```diff
@@ -1666,6 +1667,31 @@ Certificate Manager
...
+- [OpsWorks Stacks](https://aws.amazon.com/opsworks/stacks/) allows you to run your deployment stacks both in the AWS cloud as well as on your own hardware on-premises.
+- [OpsWorks for ChefAutomate](https://aws.amazon.com/opsworks/chefautomate/) provides a managed Chef configuration management server for your deployment pipeline. This server stores configuration tasks and provides them to your deployment nodes without manual intervention, plus other management and monitoring features.
```
Comment (ThanosBaskous) [inline README.md]: Phrasing suggestion: "plus other management" -> "in addition to providing other management"

File: README.md [hunk 4]
```diff
@@ -1666,6 +1667,31 @@ Certificate Manager
...
+
+### OpsWorks Alternatives and Lock-in
+
+- A few major competitors to OpsWorks include [Puppet](https://puppet.com/product/how-puppet-works) and [Ansible](https://www.ansible.com/how-ansible-works).
```
Comment (ThanosBaskous) [inline README.md]: Suggestion: Remove "A few". "Major competitors to"... and maybe note that you will have to manage / deploy them yourself.

**Comment by ClydeMachine** [comment]
Thanks @ThanosBaskous for the recommended changes! I've included them in the latest commit. I removed the "Simply put" bullet as per your recommendation - did you mean for me to remove the one previous to it as well (that compares OpsWorks with Elastic Beanstalk)? Also, I did not include any further details on the listed competitors to OpsWorks as I haven't looked into what specific differences they offer.

File: README.md [hunk 5]
```diff
@@ -1666,6 +1667,30 @@ Certificate Manager
...
+- There are numerous options, in and out of AWS, that let you automate application deployments. The separating factor between OpsWorks and other configuration management services (Elastic Beanstalk for example) is that OpsWorks specializes in letting you control the details of the systems your application runs on, where a service like Elastic Beanstalk simplifies this to focus on application configuration.
+- [OpsWorks Stacks](https://aws.amazon.com/opsworks/stacks/) allows you to run your deployment stacks both in the AWS cloud as well as on your own hardware on-premises.
+- [OpsWorks for ChefAutomate](https://aws.amazon.com/opsworks/chefautomate/) provides a managed Chef configuration management server for your deployment pipeline. This server stores configuration tasks and provides them to your deployment nodes without manual intervention, in addition to providing other management and monitoring features.
+
+### OpsWorks Alternatives and Lock-in
+
+- Major competitors to OpsWorks include [Puppet](https://puppet.com/product/how-puppet-works) and [Ansible](https://www.ansible.com/how-ansible-works).
```
Comment (jlevy) [inline README.md]: I think this bullet should say "Major competitors to Chef include..."

There could be another bullet saying "A common alternative to OpsWorks is to run Chef yourself..." (or could be covered in Basics, as commented above).

File: README.md [hunk 6]
```diff
@@ -1666,6 +1667,30 @@ Certificate Manager
 -       🔸In order to use **Certificate Manager** for CloudFront distributions certificate must be issued or imported from us-east-1 (N. Virginia) region. Certificates from other regions can [only be used with Elastic Load Balancers](https://docs.aws.amazon.com/acm/latest/userguide/acm-services.html).
 -       🔸**IoT** has its [own way](http://docs.aws.amazon.com/iot/latest/developerguide/create-device-certificate.html) of setting up certificates.
 
+OpsWorks
+-------------------
+
+### OpsWorks Basics
+
+-	📒 [Homepage](https://aws.amazon.com/opsworks/) ∙ [Documentation](https://aws.amazon.com/documentation/opsworks/) ∙ [FAQ](https://aws.amazon.com/opsworks/faqs/) ∙ [Pricing - Stacks](https://aws.amazon.com/opsworks/stacks/pricing/) ∙ [Pricing - ChefAutomate](https://aws.amazon.com/opsworks/chefautomate/pricing/)
+- OpsWorks is a configuration management service that relies heavy on Chef for configuration-as-code deployment automation. The service lets you configure and launch stacks specific to your application's needs.
```
Comment (jlevy) [inline README.md]: Best to link to Chef here.

File: README.md [hunk 7]
```diff
@@ -1666,6 +1667,30 @@ Certificate Manager
...
+- There are numerous options, in and out of AWS, that let you automate application deployments. The separating factor between OpsWorks and other configuration management services (Elastic Beanstalk for example) is that OpsWorks specializes in letting you control the details of the systems your application runs on, where a service like Elastic Beanstalk simplifies this to focus on application configuration.
```
Comment (jlevy) [inline README.md]: Would be awesome if this mentioned or linked to info on how this differs from running Chef yourself. For example, what Chef versions are supported or not? Are there any links or discussion you know of about running your own Chef vs using OpsWorks? That is the key determination someone reading this needs to make.

File: README.md [hunk 8]
```diff
@@ -1666,6 +1667,30 @@ Certificate Manager
...
+- [OpsWorks Stacks](https://aws.amazon.com/opsworks/stacks/) allows you to run your deployment stacks both in the AWS cloud as well as on your own hardware on-premises.
+- [OpsWorks for ChefAutomate](https://aws.amazon.com/opsworks/chefautomate/) provides a managed Chef configuration management server for your deployment pipeline. This server stores configuration tasks and provides them to your deployment nodes without manual intervention, in addition to providing other management and monitoring features.
+
+### OpsWorks Alternatives and Lock-in
+
+- Major competitors to OpsWorks include [Puppet](https://puppet.com/product/how-puppet-works) and [Ansible](https://www.ansible.com/how-ansible-works).
+
+### OpsWorks Tips    
+
+- OpsWorks relies heavily on Chef cookbooks and recipes, so familiarity with reading their syntax will help greatly with getting up and running.
```
Comment (jlevy) [inline README.md]: Link to docs?

**Comment by jlevy** [comment]
Leaving this open for a bit longer as we iterate on this. Also @ClydeMachine dont see you in Slack but feel free to join if you'd like to discuss more live. :)

**Comment by ClydeMachine** [comment]
Pardon the mess of commits - rebased my forked repo to keep current.

I've gone through and added several of the points you've mentioned, @jlevy - excellent points there! I didn't cover a couple of them where I didn't feel I could personally add value, and will leave for future contribution. I believe I've touched on the major edit points for this go-around.

As for Slack, thanks for the invite! I'll give it a consideration and ping you should I jump in.

**Comment by JCBarry** [comment]
This is a great section @ClydeMachine.  Thanks for adding!  A small thing, I think, is the "As of December 2016" disclaimers.  I think within reason the document can be considered "up-to-date" so it probably isn't necessary to include the timing.  If you (and/or others) feel it necessary maybe clarify the bullet with the timing at the end instead of leading with it?

Just a thought.  Thanks again! Cheers!

**Comment by ThanosBaskous** [comment]
Thanks for making those improvements, @ClydeMachine - your work to add this section is appreciated! I think that you may have accidentally picked up some other subsequent changes (including to AUTHORS.md, which we don't edit by hand). Could you unwind or in the worst case, submit a new PR? Thanks!

**Comment by QuinnyPig** [comment]
@ClydeMachine Want a hand rebasing this? Love to see this land next week.

**Comment by jlevy** [comment]
👍 Yes, agreed, this is good to go and we should figure out the git messiness. I'm good with this assuming the git messiness (resolve conflict/rebase) are sorted out. Worst case a merge that manually un-does the authors changes would be fine.

**Comment by ClydeMachine** [comment]
Pardon the delay in getting around to this - created a new PR [#425](https://github.com/open-guides/og-aws/pull/425) to undo the git history mess above, containing just the OpsWorks-related changes. Closing this PR.

**Pull request closed without merging**